File: LeetCode/7. Reverse Integer/Solution.java
public class Solution {
    public int reverse(int x) {
        int ans = 0;
        boolean flag = x >= 0;
        x = Math.abs(x);
        while (x > 0) {
            if (ans > Integer.MAX_VALUE / 10 || ans == Integer.MAX_VALUE / 10 && x > Integer.MAX_VALUE % 10 && flag || ans == Integer.MAX_VALUE && x > Integer.MAX_VALUE % 10 + 1) {
                return 0;
            }
            ans = ans * 10 + x % 10;
            x /= 10;
        }
        return flag ? ans : -ans;
    }
}

File: LeetCode/7. Reverse Integer/solution.py
class Solution(object):
    def reverse(self, x):
        """
        :type x: int
        :rtype: int
        """
        pos = x > 0
        s = int(str(abs(x))[::-1])
        if pos:
            if s > 2147483647:
                return 0
            else:
                return s
        else:
            if s > 2147483648:
                return 0
            else:
                return -s


so = Solution()
print(so.reverse(199000))
